vsm

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2017, Jaguar Land Rover
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Authors:
+#  * Gustavo Noronha <gustavo.noronha@collabora.com>
+#  * Shane Fagan <shane.fagan@collabora.com>
+
+import sys
+import argparse
+import yaml
+import ast
+
+LOGIC_REPLACE = {'||': 'or',
+                 '&&': 'and',
+                 'true': 'True',
+                 'false': 'False'}
+
+KEYWORDS = ['condition', 'emit']
+
+class State(object):
+    '''
+        Class to handle states
+    '''
+    def __init__(self, initial_state, rules):
+        class VariablesStorage(object):
+            pass
+        self.variables = VariablesStorage()
+
+        self.rules = {}
+        with open(rules) as rules_file:
+            self.parse_rules(rules_file)
+
+        if initial_state:
+            with open(initial_state) as f:
+                data = yaml.load(f.read())
+
+                for item in data:
+                    item = item.replace(" ", "").split("=")
+                    vars(self.variables)[item[0]] = item[1]
+
+    def parse_rules(self, rules_file):
+        '''
+            Parse YAML rules for policy manager and return ast code.
+        '''
+        data = rules_file.read()
+
+        # Translate logical operations to Python, so that they
+        # can be compiled.
+        for key, value in LOGIC_REPLACE.items():
+             data = data.replace(key, value).strip()
+
+        data = yaml.load(data)
```

Comment:
It'd be a good idea to use here _safe_load_ instead and possible check for any yaml scanner error and shows an appropriate error if any.